nf_core/components/remove.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from pathlib import Path
+
+import questionary
+from rich.console import Console
+from rich.rule import Rule
+from rich.syntax import Syntax
+
+import nf_core.utils
+from nf_core.components.components_command import ComponentCommand
+from nf_core.modules.modules_json import ModulesJson
+
+log = logging.getLogger(__name__)
+
+
+class ComponentRemove(ComponentCommand):
+    def __init__(self, component_type, pipeline_dir):
+        super().__init__(component_type, pipeline_dir)
+
+    def remove(self, component):
+        """
+        Remove an already installed module/subworkflow
+        This command only works for modules/subworkflows that are installed from 'nf-core/modules'
+
+        Args:
+            component (str): Name of the component to remove
+
+        Returns:
+            bool: True if any item has been removed, False if not
+        """
+        if self.repo_type == "modules":
+            log.error(f"You cannot remove a {self.component_type[:-1]} in a clone of nf-core/modules")
+            return False
+
+        # Check modules directory structure
+        self.check_modules_structure()
+
+        # Check whether pipeline is valid and with a modules.json file
+        self.has_valid_directory()
+        self.has_modules_file()
+
+        repo_dir = self.modules_repo.fullname
+        repo_path = self.modules_repo.repo_path
+        if component is None:
+            component = questionary.autocomplete(
+                f"{self.component_type[:-1]} name:",
+                choices=self.components_from_repo(repo_dir),
+                style=nf_core.utils.nfcore_question_style,
+            ).unsafe_ask()
```

Comment:
```suggestion
        repo_path = self.modules_repo.repo_path
        if component is None:
            component = questionary.autocomplete(
                f"{self.component_type[:-1]} name:",
                choices=self.components_from_repo(repo_path),
                style=nf_core.utils.nfcore_question_style,
            ).unsafe_ask()
```
repo_path contains the directory where modules are installed. Actually, I've seen that repo_dir contains `nf-core/modules`, I am wondering if we should delete the whole `self.modules_repo.fullname`, but I haven't checked if it's used somewhere else